libs/designer/src/lib/ui/panel/connectionsPanel/allConnections/connectionEntry.tsx

```diff
@@ -0,0 +1,97 @@
+import { useConnectionById } from '../../../../core/queries/connections';
+import { openPanel } from '../../../../core/state/panel/panelSlice';
+import { NodeLinkButton } from './nodeLinkButton';
+import { Icon, css } from '@fluentui/react';
+import { Button, Text, Tooltip } from '@fluentui/react-components';
+import { Open24Filled, ArrowSwap24Filled } from '@fluentui/react-icons';
+import { getConnectionErrors } from '@microsoft/utils-logic-apps';
+import { useCallback, useMemo } from 'react';
+import { useIntl } from 'react-intl';
+import { useDispatch } from 'react-redux';
+
+interface ConnectionEntryProps {
+  connectorId: string;
+  refId: string;
+  connectionReference: any;
+  brandColor?: string;
+  iconUri?: string;
+}
+
+export const ConnectionEntry = ({ connectorId, refId, connectionReference, iconUri }: ConnectionEntryProps) => {
+  const dispatch = useDispatch();
+
+  const connection = useConnectionById(connectionReference.connection.id, connectorId);
+  const nodeIds = useMemo(() => connectionReference.nodes || [], [connectionReference.nodes]);
+
+  const errors = useMemo(() => {
+    if (!connection?.result) return [];
+    return getConnectionErrors(connection?.result);
+  }, [connection]);
+
+  const intl = useIntl();
+  const openConnectionTooltipText = intl.formatMessage({
+    defaultMessage: 'Open connection',
+    description: 'Tooltip for the button to open a connection',
+  });
+  const connectedActionsText = intl.formatMessage({
+    defaultMessage: 'Connected actions',
+    description: 'Header for the connected actions section',
+  });
+  const reassignConnectionTooltipText = intl.formatMessage({
+    defaultMessage: 'Reassign all connected actions to a new connection',
+    description: 'Tooltip for the button to reassign actions',
+  });
+  // const connectionValidStatusText = intl.formatMessage({
```

Comment:
nit: do we still need this text?

Reply:
I was going to add a tooltip for the connection status with these, it was causing an error for some reason but I'll get it sorted